Error starting Micronaut server: Bean definition > DataR2dbcConfiguration

Thanks for reporting an issue, please review the task list below before submitting the
issue. Your issue report will be closed if the issue is incomplete and the below tasks not completed.

NOTE: If you are unsure about something and the issue is more of a question a better place to ask questions is on Stack Overflow (https://stackoverflow.com/tags/micronaut) or Gitter (https://gitter.im/micronautfw/). DO NOT use the issue tracker to ask questions.

### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. git clone https://github.com/edgargs/demo-r2dbc.git
2. cd demo-r2dbc
3. ./gradlew run

### Expected Behaviour

Demo R2DBC runs successful.

### Actual Behaviour

```bash
> Task :run FAILED
 __  __ _                                  _   
|  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_ 
| |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|
| |  | | | (__| | | (_) | | | | (_| | |_| | |_ 
|_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|
  Micronaut (v2.5.0)

13:56:21.553 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [io.micronaut.data.r2dbc.config.R2dbcSchemaGenerator] could not be loaded: Error instantiating bean of type  [io.micronaut.data.r2dbc.config.DataR2dbcConfiguration]

Message: Missing bean argument value: r2dbcOperations
Path Taken: new R2dbcSchemaGenerator([List configurations]) --> new DataR2dbcConfiguration(String name,ConnectionFactory connectionFactory,[R2dbcOperations r2dbcOperations])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.data.r2dbc.config.R2dbcSchemaGenerator] could not be loaded: Error instantiating bean of type  [io.micronaut.data.r2dbc.config.DataR2dbcConfiguration]

Message: Missing bean argument value: r2dbcOperations
Path Taken: new R2dbcSchemaGenerator([List configurations]) --> new DataR2dbcConfiguration(String name,ConnectionFactory connectionFactory,[R2dbcOperations r2dbcOperations])
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1565)
        at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:216)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2864)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:231)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:162)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
        at com.example.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.data.r2dbc.config.DataR2dbcConfiguration]
```

### Environment Information

- **Operating System**: Ubuntu 21.04
- **Micronaut Version:** 2.5.0
- **JDK Version:** 16

### Example Application

- https://github.com/edgargs/demo-r2dbc.git

--- a/starter-core/src/main/java/io/micronaut/starter/feature/database/r2dbc/DataR2dbc.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/database/r2dbc/DataR2dbc.java
@@ -55,7 +55,7 @@ public class DataR2dbc implements R2dbcFeature {
                 .versionProperty("micronaut.data.version")
                 .annotationProcessor());
         generatorContext.addDependency(Dependency.builder()
-                .groupId("io.micronaut.r2dbc")
+                .groupId("io.micronaut.data")
                 .artifactId("micronaut-data-r2dbc")
                 .compile());
         DatabaseDriverFeature dbFeature = generatorContext.getRequiredFeature(DatabaseDriverFeature.class);

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/database/r2dbc/DataR2dbcSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/database/r2dbc/DataR2dbcSpec.groovy
@@ -48,7 +48,7 @@ class DataR2dbcSpec extends ApplicationContextSpec implements CommandOutputFixtu
         then:
         jdbcFeature.name == 'jdbc-hikari'
         template.contains("annotationProcessor(\"io.micronaut.data:micronaut-data-processor\")")
-        template.contains('implementation("io.micronaut.r2dbc:micronaut-data-r2dbc")')
+        template.contains('implementation("io.micronaut.data:micronaut-data-r2dbc")')
         template.contains('implementation("io.micronaut.r2dbc:micronaut-r2dbc-core")')
         template.contains("runtimeOnly(\"io.r2dbc:r2dbc-h2\")")
         !template.contains("implementation(\"io.micronaut.sql:micronaut-jdbc-hikari\")")
@@ -73,7 +73,7 @@ class DataR2dbcSpec extends ApplicationContextSpec implements CommandOutputFixtu
 
         template.contains('''
     <dependency>
-      <groupId>io.micronaut.r2dbc</groupId>
+      <groupId>io.micronaut.data</groupId>
       <artifactId>micronaut-data-r2dbc</artifactId>
       <scope>compile</scope>
     </dependency>

